### DISCUSSION TITLE
Does the CheckExists method determine whether there is a delay in the existence of a job?

### DISCUSSION THREAD

**Discussion by ZLVincent** [quartznet/quartznet#1940] [Q&A] [answered]
**Describe the bug**

The results obtained by calling the CheckExists method and the GetCurrentlyExecutingJobs method are different

**Version used**

3.5.0

**Expected behavior**

I saw that the implementation of CheckExists is to judge whether the Job exists from the JOB_DETAILS table of the data. However, GetCurrentlyExecutingJobs seems to get the running Jobs from the memory. I don't know if there is a delay in updating the database. If there is a delay, will there be inconsistent results between CheckExists and GetCurrentlyExecutingJobs?

**Comment by lahma** [comment, marked as answer]
They serve a different purpose. `CheckExists` will tell if job has been registered, it might not have any triggers associated and thus might not ever be even running. `GetCurrentlyExecutingJobs` reports jobs that are currently actively executing in the specific scheduler instance.

**Reply by ZLVincent** [reply]
Thx,I understand what you mean. So for a job, if the return value of CheckExists is false, then the result returned by GetCurrentlyExecutingJobs must not include this job, right? But according to the results of my test, it is the Job in the result returned by GetCurrentlyExecutingJobs, which is judged to be false by CheckExists. That's werid. So I want to ask if it is because CheckExists queries the database, and the update of the database is delayed.

**Reply by lahma** [reply]
Two queries at different point of time can return different results, they are not part of the same transaction after all.

**Reply by ZLVincent** [reply]
ah...So these two should not be used together, Thanks a lot

**Reply by lahma** [reply]
Especially in clustered environment you probably won't get the right answer as some job could be running on different node which wouldn't be included in other node's currently executing list.

**Reply by ZLVincent** [reply]
Then in the cluster environment, how to get the currently executing list?

**Reply by lahma** [reply]
There's no API for that which would report entire cluster state.
